src/NuGet.Clients/VsConsole/PowerShellHost/RunspaceManager.cs

```diff
@@ -53,13 +53,18 @@ internal class RunspaceManager : IRunspaceManager
             DTE dte = ServiceLocator.GetInstance<DTE>();
```

Comment:
This query for DTE should probably also be while you're on the UI thread. Can you move this line into your JTF.Run delegate under your SwitchToMainThread?
It's not an issue if you're certain that your ServiceLocator *already has cached* this value and this method only returns the value (without casting it). But I'd still make the change so it's clearly correct rather than only correct after deep thinking about where the value comes from.